tests/theme/dark-mode.spec.tsx

```diff
@@ -0,0 +1,191 @@
+import React from "react";
+import { render } from "@testing-library/react";
+import styled from "styled-components";
+import {
+    Colour,
+    DSThemeProvider,
+    LifeSGTheme,
+    useDSTheme,
+} from "../../src/theme";
+
+// Test component that uses theme colors
+const TestComponent = styled.div`
+    background: ${Colour["bg"]};
+    color: ${Colour["text"]};
+`;
+
+// Component to test the useDSTheme hook
+const ThemeModeDisplay = () => {
+    const { colourMode } = useDSTheme();
+    return <div data-testid="theme-mode">{colourMode}</div>;
+};
+
+describe("Dark Mode Theme", () => {
```

Comment:
we should also test the system preference behaviour

may need to mock `window.matchMedia` to return the expected values and trigger the change events, a possible method can be found [here](https://stackoverflow.com/questions/39830580/jest-test-fails-typeerror-window-matchmedia-is-not-a-function)